internal/js/modules/k6/browser/common/frame_manager.go

```diff
@@ -540,7 +540,32 @@ func (m *FrameManager) requestStarted(req *Request) {
 		frame.pendingDocument = &DocumentInfo{documentID: req.documentID, request: req}
 		frame.pendingDocumentMu.Unlock()
 	}
-	m.logger.Debugf("FrameManager:requestStarted", "fmid:%d rurl:%s pdoc:nil", m.ID(), req.URL())
+
+	if !m.page.hasRoutes() {
+		return
+	}
+
+	route := NewRoute(m.logger, m.page.mainFrameSession.networkManager, req)
+	for _, r := range m.page.routes {
```

Comment:
Do you need to protect this read with the `routesMu`?

Reply:
Fixed in https://github.com/grafana/k6/pull/4953/commits/ae4f370acb8fa6e131864aedae0ba9dae50f6491